test/new-e2e/tests/windows/service-test/startstop_test.go

```diff
@@ -671,6 +671,12 @@ func (s *baseStartStopSuite) assertServiceState(expected string, serviceName str
 			s.T().Logf("waiting for %s to be %s, status %s", serviceName, expected, status)
 		}
 	}, 1*time.Minute, 1*time.Second, "%s should be in the expected state", serviceName)
+
+	// if a driver service failed to get to the expected state, capture a kernel dump for debugging.
+	if s.T().Failed() && slices.Contains(s.getInstalledKernelServices(), serviceName) {
+		s.T().Logf("capturing live kernel dump due to %s not in %s state\n", serviceName, expected)
+		s.captureLiveKernelDump(host, s.SessionOutputDir())
+	}
```

Comment:
Do we need to capture dump always in bad states. I think that if a driver failed to start then no, but pending start or stop always and running when it should be stopped.